control/delaylti.py

```diff
@@ -0,0 +1,842 @@
+        -----
+        Two DelayLTI systems are considered equal if their underlying
+        PartitionedStateSpace representations are equal and their delay
+        vectors (`tau`) are element-wise identical.
+        """
+        if not isinstance(other, DelayLTI):
+            raise TypeError(f"{other} is not a DelayLTI object, is {type(other)}")
+        return (self.P == other.P) and (self.tau == other.tau)
+
+    def feedback(self, other=1, sign=-1):
+        """Standard or LFT feedback interconnection for DelayLTI.
+
+        If `other` is a static gain (scalar, matrix, or static LTI system),
+        computes the standard feedback loop: u = r + sign*other*y, y = self*u.
+        The resulting system maps the external input `r` and the internal
+        delay input `w` to the external output `y` and the internal delay
+        output `z`.
+
+        If `other` is also a `DelayLTI`, computes the LFT feedback
+        interconnection by calling `feedback` on the underlying
+        `PartitionedStateSpace` objects and concatenating the delay vectors.
+
+        Parameters
+        ----------
+        other : scalar, array, LTI system, or DelayLTI
+            The system or gain in the feedback path.
+        sign : int, optional {-1, 1}
+            Sign of the feedback. Default is -1 (negative feedback).
+
+        Returns
+        -------
+        DelayLTI
+            The closed-loop system.
+        """
+
+        if isinstance(other, DelayLTI):
+            psys_new = self.P.feedback(other.P)
+            tau_new = np.concatenate([self.tau, other.tau])
+            return DelayLTI(psys_new, tau_new)
+
+        elif isinstance(other, (StateSpace, TransferFunction)):
+            other_delay_lti = _convert_to_delay_lti(other)
+            return self.feedback(other_delay_lti)
+
+        else:
+            # Convert feedback 'other' to a static gain matrix K
+            if isinstance(other, (int, float, complex, np.number)):
+                if not self.issiso():
+                    raise ValueError("Scalar feedback gain requires SISO system G.")
+                K = np.array([[other]], dtype=float)
+            elif isinstance(other, np.ndarray):
+                K = np.asarray(other, dtype=float)
+                if K.ndim == 0:
+                    K = K.reshape(1, 1)
+                elif K.ndim == 1:
+                    if self.nu != 1:
+                        raise ValueError("1D array feedback requires SISO system G.")
+                    K = K.reshape(self.ninputs, 1)
+                elif K.ndim != 2:
+                    raise ValueError("Feedback gain must be scalar, 1D, or 2D array.")
+            else:
+                raise TypeError(f"Unsupported type for static feedback: {type(other)}")
+
+            # Check dimensions of K
+            if K.shape != (self.nu, self.ny):
+                raise ValueError(
+                    f"Feedback gain K has incompatible shape. Expected ({self.nu}, {self.ny}), got {K.shape}."
```

Comment:
Update to avoid line with more than 78 characters and put closing parenthesis at end of string, for consistency with current code base.